### ISSUE TITLE
Complex maths kills the launcher.

### ISSUE THREAD

**Issue by WinnerWind** [MM2-0/Kvaesitso#1034]
When running complex maths equations that can't be solved by normal means, the launcher tends to crash

Eg. Running `3^3^3^3` in the search menu, the launcher crashes. Same goes for `pi^pi^pi^pi`. These numbers are trillions of digits long and cannot be calculated. 

Mathematically these are simply accurately incalculable and hence Kvaesitso should choose to ignore the request, throw an error message, or silently switch to a scientific calculator (eg. 778193749183729 can be approximated to 7.78*10¹³ to save computer resources, even if it's not exact. This should prevent crashes, but in the above case, it may still crash)

I was not able to find any other equations that were not power chains.

Proof that this only affects incalculable solutions
`2^2^2^2` does compute properly, so this is not an issue with the algorithm that does the powers.

Edit : 3^3^33 does crash, but 33^333 returns infinity. I wonder why this is the case. Does your arithmetic library automatically filter these out?

**Comment by MM2-0** [comment]
Can you post the crash report? (Settings > Debug > Crash reporter)

**Comment by WinnerWind** [comment]
[kvaesitso-log-2024-08-15-203141.txt](https://github.com/user-attachments/files/16626641/kvaesitso-log-2024-08-15-203141.txt)
[2024-08-15 20_31_29_crash.txt](https://github.com/user-attachments/files/16626644/2024-08-15.20_31_29_crash.txt)
```
APP.VERSION : 1.32.2
APP.VERSIONCODE : 2024072400
APP.SIGNATURE : rx1fSnL7r5/OMoFC0e1KPqTndXQ=
LAUNCHER.APP : android
TIMEZONE : Asia/Calcutta
VERSION.RELEASE : 13
VERSION.INCREMENTAL : V816.0.1.0.TKCINXM
VERSION.SDK.NUMBER : 33
BOARD : veux
BOOTLOADER : unknown
BRAND : Redmi
CPU_ABI : arm64-v8a
CPU_ABI2 : 
DISPLAY : TKQ1.221114.001 test-keys
FINGERPRINT : Redmi/peux_in/peux:13/TKQ1.221114.001/V816.0.1.0.TKCINXM:user/release-keys
HARDWARE : qcom
HOST : pangu-build-component-system-75948-sdqdq-n26kd-7rkch
ID : TKQ1.221114.001
MANUFACTURER : Xiaomi
MODEL : 2201116SI
PRODUCT : peux_in
TAGS : release-keys
TIME : 1712064303000
TYPE : user
```
tldr seems to be a java arithmetic issue. Quite common, but to my knowledge can be avoided by sandboxing the arithmetic operation (so if it crashes the launcher doesnt go AWOL) or by adding a filter before the operation is run so that it doesn't do large compute cycles like this.

**Comment by MM2-0** [comment]
Thanks. Should be trivial to catch.

> Apologies, it seems that I forgot to add the "bug" tag to this issue request

No worries lol

**Comment by WinnerWind** [comment]
Great stuff! I would love to test it, but considering how minor this issue is and I have no idea how to build and sign android apps, I'll wait for the next release. Thanks for fixing it!

**Comment by MM2-0** [comment]
You could try the nightly version, but it's probably not worth it. After all, I tested it myself, and what more is there to test.
